public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -176,16 +188,21 @@ export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
       dashboard.meta.fromExplore = !!(panelId && queries);
     }
 
-    // template values service needs to initialize completely before
-    // the rest of the dashboard can load
+    // template values service needs to initialize completely before the rest of the dashboard can load
     await dispatch(initVariablesTransaction(args.urlUid, dashboard, variableSrv));
+
     if (getState().templating.transaction.uid !== args.urlUid) {
       // if a previous dashboard has slow running variable queries the batch uid will be the new one
       // but the args.urlUid will be the same as before initVariablesTransaction was called so then we can't continue initializing
       // the previous dashboard.
       return;
     }
 
+    // If dashboard is in a different init phase it means it cancelled during service init
+    if (getState().dashboard.initPhase !== DashboardInitPhase.Services) {
+      return;
+    }
```

Comment:
Added this as way to not issue dashboardInitCompleted action if the dashboard has been cleanup during init.